odoo/osv/expression.py

```diff
@@ -711,6 +711,14 @@ def push_result(query, params):
                 subquery, subparams = query.select('"%s"."%s"' % (comodel._table, field.inverse_name))
                 push(('id', 'inselect', (subquery, subparams)), model, alias, internal=True)
 
+            elif field.type in ('json', 'properties'):
+                assert len(path) == 2
+                assert regex_pg_name.match(path[1]), "Wrong property name"
+                assert operator in ('=', '!=', 'like', 'ilike')
```

Comment:
What about '=ilike' ? Could we support it?

Does this mean when searching "if m2m contains", we should use ilike and not "in"?
Will be strange if we are trying to find ID 42 and the value contains "[142, 342, 85]", will return True when it's not ^^
(Same logic for your special 'tags' one, may not work properly).

Isn't there a way to search "if a json array contains" instead for those?

That would require checking the "type" of the sub-operator though, not sure if possible.

If you add those, will probably require more tests.

Reply:
> Isn't there a way to search "if a json array contains" instead for those?

Yes, I wrote a comment above about that (with a stackoverflow link, it's possible but it would require a bit of logic

For the tags it shouldn't be that hard to do (but take some times, so I'll check when we will try to add it in the search view)

> That would require checking the "type" of the sub-operator though, not sure if possible.

Looks like postgre is just waiting a string, (see https://stackoverflow.com/questions/20236421/how-to-convert-postgres-json-to-integer)

Search work on number with this

But for the many2one / many2many it will be super tricky... I don't even know if it would be possible... (would need to do 2 search, one to get the ids based on the name and the second to search in the JSON... or with a JOIN query...)

Because it's super tricky (and error prone, the smallest error will introduce a general SQL injection in Odoo) I propose to let it as it is, and see if we have the time to implement it once the rest is done